File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/单调栈/LeetCode_0739.java
package com.kurumi.codeThoughts.单调栈;

import java.util.Stack;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description LeetCode_0739
 */
public class LeetCode_0739 {
    static class Solution {
        public int[] dailyTemperatures(int[] temperatures) {
            int[] res = new int[temperatures.length];
            Stack<Integer> st = new Stack<>();
            for (int i = 0; i < temperatures.length; i++) {
                while (!st.empty() && temperatures[i] > temperatures[st.peek()]) {
                    res[st.peek()] = i - st.peek();
                    st.pop();
                }
                st.push(i);
            }
            return res;
        }
    }
}

File: 卡码/代码随想录/C++/单调栈/monotonicStack/LeetCode_0739.cpp
//739. 每日温度
#include <iostream>
#include <stack>
#include <vector>
using namespace std;

class Solution {
public:
	vector<int> dailyTemperatures(vector<int>& temperatures) {
		stack<int> st;
		vector<int> res(temperatures.size(),0);
		for(int i=0;i<temperatures.size();i++){
			while(!st.empty()&&temperatures[i]>temperatures[st.top()]){
				res[st.top()]=i-st.top();
				st.pop();
			}
			st.push(i);
		}
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}
